core/services/fluxmonitorv2/flux_monitor_test.go

```diff
@@ -0,0 +1,1674 @@
+	chSafeToFillQueue := make(chan struct{})
+
+	pipelineRunner := new(pipelinemocks.Runner)
+	pipelineSpec := NewPipelineSpec()
+	l := *logger.Default
+
+	pipelineRun := fluxmonitorv2.NewPipelineRun(
+		pipelineRunner,
+		pipelineSpec,
+		spec.JobID,
+		l,
+	)
+
+	fluxAggregator := new(mocks.FluxAggregator)
+	fluxAggregator.On("LatestRoundData", nilOpts).Return(freshContractRoundDataResponse()).Once()
+	fluxAggregator.On("OracleRoundState", nilOpts, nodeAddr, uint32(1)).
+		Return(makeRoundStateForRoundID(1), nil).
+		Run(func(mock.Arguments) {
+			close(chSafeToFillQueue)
+			<-chBlock
+		}).
+		Once()
+	fluxAggregator.On("OracleRoundState", nilOpts, nodeAddr, uint32(3)).Return(makeRoundStateForRoundID(3), nil).Once()
+	fluxAggregator.On("OracleRoundState", nilOpts, nodeAddr, uint32(4)).Return(makeRoundStateForRoundID(4), nil).Once()
+	fluxAggregator.On("GetOracles", nilOpts).Return(oracles, nil)
+	fluxAggregator.On("Address").Return(spec.ContractAddress, nil)
+
+	logBroadcaster := new(logmocks.Broadcaster)
+	logBroadcaster.On("Register", mock.Anything, mock.Anything).Return(true)
+	logBroadcaster.On("Unregister", mock.Anything, mock.Anything)
+
+	fmstore.On("MostRecentFluxMonitorRoundID", spec.ContractAddress).Return(uint32(1), nil)
+	fmstore.On("MostRecentFluxMonitorRoundID", spec.ContractAddress).Return(uint32(3), nil)
+	fmstore.On("MostRecentFluxMonitorRoundID", spec.ContractAddress).Return(uint32(4), nil)
+
+	// Round 1
+	fmstore.
+		On("FindOrCreateFluxMonitorRoundStats", spec.ContractAddress, uint32(1)).
+		Return(fluxmonitorv2.FluxMonitorRoundStatsV2{
+			Aggregator: spec.ContractAddress,
+			RoundID:    1,
+		}, nil)
+	pipelineRunner.
+		On("ExecuteAndInsertNewRun", context.Background(), pipelineSpec, l).
+		Return(int64(1), pipeline.FinalResult{
+			Values: []interface{}{decimal.NewFromInt(fetchedValue)},
+			Errors: []error{nil},
+		}, nil)
+	fluxAggregator.On("Submit",
+		&bind.TransactOpts{
+			From: transmissionAddress,
+		},
+		big.NewInt(1),
+		big.NewInt(fetchedValue),
+	).Return(&types.Transaction{}, nil)
+	fmstore.
+		On("UpdateFluxMonitorRoundStats",
+			spec.ContractAddress,
+			uint32(1),
+			mock.AnythingOfType("int64"), //int64(1),
+		).
+		Return(nil).Once()
+
+	// Round 3
+	fmstore.
+		On("FindOrCreateFluxMonitorRoundStats", spec.ContractAddress, uint32(3)).
+		Return(fluxmonitorv2.FluxMonitorRoundStatsV2{
+			Aggregator: spec.ContractAddress,
+			RoundID:    3,
+		}, nil)
+	pipelineRunner.
+		On("ExecuteAndInsertNewRun", context.Background(), pipelineSpec, l).
+		Return(int64(2), pipeline.FinalResult{
+			Values: []interface{}{decimal.NewFromInt(fetchedValue)},
+			Errors: []error{nil},
+		}, nil)
+	fluxAggregator.On("Submit",
+		&bind.TransactOpts{
+			From: transmissionAddress,
+		},
+		big.NewInt(3),
+		big.NewInt(fetchedValue),
+	).Return(&types.Transaction{}, nil)
+	fmstore.
+		On("UpdateFluxMonitorRoundStats",
+			spec.ContractAddress,
+			uint32(3),
+			mock.AnythingOfType("int64"), //int64(2),
```

Comment:
Because the calls can come in different orders, I wasn't able to determine the run id consistently. 

Is there a better way to handle these mocks?